src/pages/iou/MoneyRequestModal.js

```diff
@@ -191,11 +208,30 @@ const MoneyRequestModal = (props) => {
         // eslint-disable-next-line react-hooks/exhaustive-deps -- props does not need to be a dependency as it will always exist
     }, [currentStepIndex, props.translate]);
 
+    /**
+     * Navigate to a provided step.
+     *
+     * @param {Number} stepIndex
+     * @type {(function(*): void)|*}
+     */
+    const navigateToStep = useCallback((stepIndex) => {
+        if (stepIndex < 0 || stepIndex > steps.length) {
+            return;
+        }
+
+        if (currentStepIndex === stepIndex) {
+            return;
+        }
+
+        setPreviousStepIndex(currentStepIndex);
+        setCurrentStepIndex(stepIndex);
+    }, [currentStepIndex, steps.length]);
+
     /**
      * Navigate to the previous request step if possible
      */
     const navigateToPreviousStep = useCallback(() => {
-        if (currentStepIndex <= 0) {
+        if (currentStepIndex <= 0 && previousStepIndex < 0) {
```

Comment:
I think that we need to update the dependency array for the `useCallback` hook to that it returns the updated memoized function